frontend/utils/distribution/monthly-event-parser.js

```diff
@@ -0,0 +1,226 @@
+      const deltaCycle = (cyclesIntoMonth - member.cyclicalIncomeVariable)
+      // Update the existing user's pro-rated income (cyclicalIncomeIntegral), time-variable (cyclicalIncomeVariable), and currently declared income:
+      member.cyclicalIncomeVariable = cyclesIntoMonth
+      member.cyclicalIncomeIntegral = monthlyRated ? member.haveNeed : member.cyclicalIncomeIntegral + deltaCycle * member.haveNeed
+      member.haveNeed = member.cyclicalIncomeIntegral
+    }
+    return proRatedMembers
+  }
+
+  const redistributOverToLatePayments = function (overPayments, latePayments) {
+    const adjustingPayments = []
+    const needers = groupMembers.filter((m) => m.haveNeed < 0)
+    const totalNeed = needers.reduce((acc, m) => acc + m.haveNeed, 0)
+    for (const overPayment of overPayments) {
+      for (const needer of needers) {
+        if (needer.name !== overPayment.to) {
+          adjustingPayments.push({
+            from: overPayment.from,
+            to: needer.name,
+            amount: overPayment.amount * needer.haveNeed / totalNeed
+          })
+        }
+      }
+    }
+    return reduceDistribution([latePayments, adjustingPayments].flat())
+  }
+
+  // Make a place to store the previous cycle's startCycleEvent (where over/under-payments are stored)
+  // so that they can be included in the next cycle's payment distribution calculations:
+  let lastStartCycleEvent = distributionEvents[0]
+  let monthlyDistribution = [] // For each cycle's monthly distribution calculation
+  let completedMonthlyPayments = [] // For accumulating the payment events of each month's cycle.
+
+  // Create a helper function for calculating each cycle's payment distribution:
+  const paymentsDistribution = function (groupMembers, minCome) {
+    const groupIncomes = groupMembers.map((user) => {
+      return {
+        name: user.name,
+        amount: minCome + user.haveNeed
+      }
+    })
+    return incomeDistribution(groupIncomes, minCome)
+  }
+  // Loop through the events, pro-rating each user's monthly pledges/needs:
+  for (const event of distributionEvents) {
+    if (event.type === 'startCycleEvent') {
+      monthlyDistribution = paymentsDistribution(proRateHaveNeeds(groupMembers), minCome)
+      // "Double-Adjust" the monthly distribution based on the current cycle's completed payments, and
+      // the previous cycle's under payments:
+      monthlyDistribution = addDistributions(
+        subtractDistributions(monthlyDistribution, completedMonthlyPayments),
+        lastStartCycleEvent.data.latePayments)
+
+      const overPayments = monthlyDistribution.filter((p) => {
+        return p.amount < 0
+      }).map((p) => {
+        p.amount = Math.abs(p.amount)
+        return p
+      })
+
+      const latePayments = monthlyDistribution.filter((p) => {
+        return p.amount > 0
+      })
+
+      lastStartCycleEvent = event
+      lastStartCycleEvent.data.latePayments = redistributOverToLatePayments(overPayments, latePayments)
```

Comment:
This causes a Vue error (the one we talked about on the call), just marking this for reference (since the data passed in is being modified...)

Reply:
`redistributOverToLatePayments ` has been removed in later updates, and there don't seem to be any Vue errors in the latest.

Marking resolved, for now.